### PR TITLE
MODCONF-33: Use tenant API for loading sample data

### PR DESCRIPTION
**Pull request by KVupp**
## PURPOSE
https://issues.folio.org/browse/MODCONF-33

## APPROACH
- Loading sample data using the tenant API
- Added unit test cases for the same

Things to Note:
- While working on this story observed that PUT /configurations/entries throws 500 on entity not found instead of 404. 
The TenantLoading class (https://github.com/folio-org/raml-module-builder/blob/master/domain-models-runtime/src/main/java/org/folio/rest/tools/utils/TenantLoading.java#L286) looks for 404, and then tries to do a POST-- it exits on receiving a 500 hence  added a parameter withPostOnly, so that it always tries to do a POST to load the sample data.
This has to be removed once the bug is fixed(https://issues.folio.org/browse/MODCONF-35)

## TODO
- [x] Inform infrastructure team to change the load parameter for this module to add loadSample=true
- [x] If this story is merged before https://issues.folio.org/browse/MODCONF-35, make sure to add a note in the story to remove withPostOnly from the TenantReferenceAPI class

### CODE REVIEW

File: mod-configuration-server/src/main/resources/data/locales/currency_code_dk.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "id": "b873eb5a-7a50-488a-9624-d4fbc4daad51",
```
Comment (grabsefx) [inline mod-configuration-server/src/main/resources/data/locales/currency_code_dk.json]: I see that indentations differ in all samples. Please, reformat files in folder /data/locales.

File: mod-configuration-server/src/test/java/org/folio/rest/RestVerticleTest.java [hunk 2]
```diff
@@ -144,6 +138,56 @@ public void beforeEach()
...
 
+
+  @Test
+  public void verifySampleDataLoaded(TestContext testContext) {
+    final Async async = testContext.async();
+    okapiHttpClient.get("http://localhost:" + port + "/configurations/entries?query=module==SETTINGS")
+  .thenAccept(response -> {
+    try {
+      testContext.assertEquals(200, response.getStatusCode(),
+        String.format("Unexpected status code: '%s': '%s'", response.getStatusCode(),
```
Comment (grabsefx) [inline mod-configuration-server/src/test/java/org/folio/rest/RestVerticleTest.java]: This formatted string value repeats in many places. You could move it to a constant string and just reuse it.

File: mod-configuration-server/src/main/resources/data/locales/currency_code_dk.json (hunk 1 above)
Comment (KVupp) [reply mod-configuration-server/src/main/resources/data/locales/currency_code_dk.json]: done

File: mod-configuration-server/src/test/java/org/folio/rest/RestVerticleTest.java (hunk 2 above)
Comment (KVupp) [reply mod-configuration-server/src/test/java/org/folio/rest/RestVerticleTest.java]: done

**Comment by KVupp** [comment]
https://github.com/folio-org/folio-ansible/pull/283  -- PR lo load sample data in reference environments

**Pull request merged**